src/task_scl_structural_habitat.py

```diff
@@ -32,337 +37,49 @@ def __init__(self, *args, **kwargs):
         )
         self.elevation = ee.Image(self.inputs["elevation"]["ee_path"])
         self.zones = ee.FeatureCollection(self.inputs["zones"]["ee_path"])
-        # self.set_aoi_from_ee(self.inputs["probability"]["ee_path"])
+        self.lc_elev_reclass = ee.FeatureCollection(
+            self.inputs["lc_elev_reclass"]["ee_path"]
+        )
 
     def calc(self):
 
-        zones_img = self.zones.reduceToImage(
-            properties=['Biome_zone'],
-            reducer=ee.Reducer.mode()
-        ).rename('Biome_zone')
-
-        def remap_elev(from_list, to_list, zone_number, z_img, elev_img, lc_img):
-            return (elev_img.lt(
-            lc_img.remap(from_list, to_list)
-                .updateMask(z_img.eq(zone_number)))
-                .multiply(100))
-
+        zone_numbers = self.zones.aggregate_histogram("Biome_zone").keys()
+        lc_val = self.lc_elev_reclass.aggregate_array("lc_value")
 
-        def remap_lc_by_zone(lc_list, z1_list, z2_list, z3_list, z_img, elev_img, lc_img):
-            lc_zone1=remap_elev(lc_list, z1_list, 1,
-                                zones_img, elev_img, lc_img)
-            lc_zone2=remap_elev(lc_list, z2_list, 2,
-                                zones_img, elev_img, lc_img)
-            lc_zone3=remap_elev(lc_list, z3_list, 3,
-                                zones_img, elev_img, lc_img)
-            return (ee.ImageCollection.fromImages([lc_zone1, lc_zone2, lc_zone3])
-                .mosaic()
+        zones_img = self.zones.reduceToImage(
+            properties=["Biome_zone"], reducer=ee.Reducer.mode()
+        ).rename("Biome_zone")
+
+        def str_hab_by_zone(li):
+            zone_string = ee.String(li)
+            zone_number = ee.Number.parse(li)
+            column = ee.String("elev_zone").cat(zone_string)
+            elev_zone = self.lc_elev_reclass.aggregate_array(column)
+            reclass_img = (
+                self.elevation.lt(self.land_cover.remap(lc_val, elev_zone))
```

Comment:
wonder if we want `elevation.lte`?

Reply:
probably. will confirm and change in necessary.